instructions/instructions_impl.cpp

```diff
@@ -4,87 +4,123 @@
 #include <string>
 #include <vector>
 
+/*  Implementation of instructions   */
+
+/*  Constructors check arguments and can throw ParserException.
+    In this case, the fields have not yet been validated and a destructor is not needed. */
+
+
+Add::Add(vector<string> args) {
+  // converted to types: Register, Register, Register
+  int args_amount = 3;
+  if (args.size() != args_amount) {
+    throw ParserException("add", args_amount, args.size());
+  }
+  Register dist_ = Parser::get_register(args[0]);
+  Register source1_ = Parser::get_register(args[1]);
+  Register source2_ = Parser::get_register(args[2]);
+
+  dist = dist_;
+  source1 = source1_;
+  source2 = source2_;
+}
+
 void Add::exec(State &state) {
   if (dist == zero) {
     return;
   }
   state.registers[dist] = state.registers[source1] + state.registers[source2];
 }
 
-void Add::fill_args(vector<string> args) {
-  // converted to types: Register, Register, Register
-  int args_amount = 3;
+
+Li::Li(vector<string> args) {
+  // converted to types: Register, int
+  int args_amount = 2;
   if (args.size() != args_amount) {
-    throw ParserException(
-        Parser::exception_message("add", args_amount, args.size()));
+    throw ParserException("li", args_amount, args.size());
+  }
+  Register dist_ = Parser::get_register(args[0]);
+  int immediate_;
+  if (Parser::is_number(args[1])) {
+      immediate_ = stoi(args[1]);
+  } else {
+    throw ParserException("invalid immediate in li: " + args[1]);
```

Comment:
You change ParserException constructor right?

Reply:
Yes, I moved the function from Parser to the ParserException constructor. I wrote about all the changes in the description of pr, you can check it)